Using-GTSAM-EXPORT.md

```diff
@@ -0,0 +1,41 @@
+# Using GTSAM_EXPORT
+
+To create a DLL in windows, the `GTSAM_EXPORT` keyword has been created and needs to be applied to different methods and classes in the code to expose this code outside of the DLL.  However, there are several intricacies that make this more difficult than it sounds.  In general, if you follow the following three rules, GTSAM_EXPORT should work properly.  The rest of the document also describes (1) the common error message encountered when you are not following these rules and (2) the reasoning behind these usage rules.
+
+## Usage Rules
+1.  Put `GTSAM_EXPORT` in front of any function that you want exported in the DLL _if and only if_ that function is defined in a .cpp file, not just a .h file.
```

Comment:
I don't totally get this one... perhaps you refer here to inlined functions defined in headers?
Also, this rule may get confusing when working with templates, explicitly instantiated templates, etc. which may be declared in the .h, and defined (and "exported") in the .cpp, etc. 

What's a clear use example you have in your mind when writing this rule?

Reply:
A couple of examples:

1.  inline_QR did not have the GTSAM_EXPORT keyword in front of it.  It was declared in a .cpp file, but not in the .h file, so there was a linker error when another library called inline_QR.
2.  The Lie* classes did not have any functions declared in a .cpp file.  If you added GTSAM_EXPORT to the class definition, you got errors when calling any of those functions because they were _not_ in the DLL.

My understanding is basically if it is completely declared in a header, there is no reason to put it in the DLL.

I am wondering if the confusion may be that I put the wrong word in.  It should really be "that function is _declared_" not _defined_.

Does that make sense?  I have changed the defined to declared in the .md file.